src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs

```diff
@@ -215,6 +215,12 @@ private static PerLanguageOption2<CodeStyleOption2<bool>> CreateQualifyAccessOpt
             "dotnet_style_prefer_is_null_check_over_reference_equality_method",
             $"TextEditor.%LANGUAGE%.Specific.{nameof(PreferIsNullCheckOverReferenceEqualityMethod)}");
 
+        internal static readonly PerLanguageOption2<CodeStyleOption2<bool>> PreferParameterNullChecking = CreateOption(
+            CodeStyleOptionGroups.ExpressionLevelPreferences, nameof(PreferParameterNullChecking),
+            defaultValue: TrueWithSuggestionEnforcement,
+            "dotnet_style_prefer_parameter_null_checking",
+            $"TextEditor.%LANGUAGE%.Specific.{nameof(PreferParameterNullChecking)}");
```

Comment:
probably not done yet.  but we need to update the VS options UI to show this.  @jmarolf do we need to updat ethe editorconfig UI as well?

Reply:
@CyrusNajmabadi I've been thinking for veeery loong to have a source generator help in this area. Do you think this is something interesting? I may try to work on it in free time.

Reply:
Works for me

Reply:
I am going through the boilerplate now and yeahhh I would love for a source generator to make all this go away.